### PR TITLE
Rmc jij cycle fix

### PR DESCRIPTION
**Pull request by rclough**
for #172

### CODE REVIEW

File: dagobah/core/job.py [hunk 1]
```diff
@@ -639,7 +642,8 @@ def verify(self, context=None):
             verified = cur_job.verify(context)
             if not verified:
                 logger.warn("Cycle or error detected in sub-job: {0}"
-                            .format(cur_job))
+                            .format(cur_job.name))
                 return False
 
+        context.discard(self.name)
```
Comment (rclough) [inline dagobah/core/job.py]: this is the main fix, the others are just better logging

**Comment by kristiankaufmann** [comment]
maybe change the name of `context` to something descriptive ... perhaps `job_names`?

Also are there unit tests for this function?

**Comment by rclough** [comment]
The context is actually in some sense more descriptive than "job names" in regards to the expansion process as a whole, as it describes the *context* under which the current job is expanding- ie the jobs it has already expanded from, which allows it to detect if a cycle would occur.

**Comment by rclough** [comment]
not that I'd be opposed to a name change, but I think `job_names` is a bit underdescriptive, but not sure what else would make sense and also be a succinct enough variable, ie `jobs_expanded_from` is a bit verbose, I think

**Comment by kristiankaufmann** [comment]
How about `ancestor_jobs`?

**Comment by rclough** [comment]
I say :+1: to that. for it being descriptive and succinct. I'll work on a separate PR for that.

**Comment by rclough** [comment]
Gonna add a unit test for this case in the meantime

**Comment by rclough** [comment]
Unit test added!

File: tests/test_core.py [hunk 2]
```diff
@@ -512,6 +512,23 @@ def test_dagobah_expand_simple_job():
 
 
 @with_setup(blank_dagobah)
+def test_dagobah_expand_same_job_twice():
+    dagobah.add_job('test_job_a')
```
Comment (kristiankaufmann) [inline tests/test_core.py]: It looks like you are testing successful conditions. I think you should also test some failure conditions as well.

Comment (rclough) [inline tests/test_core.py]: Failure conditions for what? This fix is to fix a false negative, ie the job fails when it is supposed to pass, so I tested for the most basic case. I'm not sure how I'd test this issue in an inverse or converse manner, without repeating existing tests.

There are already a few tests to ensure that job expansion fails when it is supposed to fail:

* https://github.com/rclough/dagobah/blob/rmc-JIJ-cycle-fix/tests/test_core.py#L388
* https://github.com/rclough/dagobah/blob/rmc-JIJ-cycle-fix/tests/test_core.py#L402

**Comment by kristiankaufmann** [comment]
works for me then

**Comment by ikiini** [comment]
Looks good to me. :+1:

**Comment by davidchen322** [comment]
looks good

**Pull request merged**